db/attachment.go

```diff
@@ -500,14 +472,30 @@ func GenerateProofOfAttachment(attachmentData []byte) (nonce []byte, proof strin
 
 //////// HELPERS:
 
-func BodyAttachments(body Body) map[string]interface{} {
-	atts, _ := body["_attachments"].(map[string]interface{})
-	return atts
+func BodyAttachments(body Body) (attachmentMap AttachmentMap) {
+	switch a := body["_attachments"].(type) {
+	case AttachmentMap:
+		attachmentMap = a
+	case map[string]interface{}:
+		base.DeepCopyInefficient(&attachmentMap, a)
```

Comment:
We probably want a custom converter from map[string]interface{} to AttachmentMap (i.e. pulling out the attributes we're interested in) instead of the marshal/unmarshal approach used by DeepCopyInefficient.